tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SymbStateDecoder.scala

```diff
@@ -75,21 +76,40 @@ class SymbStateDecoder(solverContext: SolverContext, rewriter: SymbStateRewriter
     case IntT() =>
       solverContext.evalGroundExpr(cell.toNameEx).withTag(Typed(IntT1()))
 
-    case ConstT() =>
-      val found = rewriter.strValueCache.findKey(cell)
-      if (found.isDefined) {
-        tla.str(found.get).typed()
-      } else {
-        findCellInSet(arena, rewriter.strValueCache.values().toSeq, cell.toNameEx) match {
-          // found among the cached keys
-          case Some(c) =>
-            decodeCellToTlaEx(arena, c)
-
-          case None =>
-            // not found, just use the name
-            // a value that was assigned by the solver, and not created by us
-            tla.str(cell.toString).typed()
-        }
+    case ConstT(utype) =>
+      utype match {
+        case "" =>
+          val found = rewriter.strValueCache.findKey(cell)
+          if (found.isDefined) {
+            tla.str(found.get).typed()
+          } else {
+            findCellInSet(arena, rewriter.strValueCache.values().toSeq, cell.toNameEx) match {
```

Comment:
What is the difference between `rewriter.strValueCache.findKey` and `findCellInSet`? A comment would help.

Reply:
This is literally unchanged code for the string case, you wrote this part.

Reply:
I have no idea about what it is doing. Worth writing a comment, if you understood it? :))